cli/pcluster/config/param_types.py

```diff
@@ -675,6 +683,59 @@ def to_file(self, config_parser):
         pass
 
 
+class DisableHyperThreading(Param):
+    """
+    Class to manage the disable_hyperthreading configuration parameter.
+
+    We need this class in order to convert the boolean disable_hyperthreading = [true/false] into Cores.
+    """
+
+    def from_cfn_params(self, cfn_params):
+        """Initialize param value by parsing the right CFN input."""
+        cfn_converter = self.definition.get("cfn_param_mapping", None)
+        if cfn_converter and cfn_params:
+            cores = int(get_cfn_param(cfn_params, cfn_converter).split(",")[0])
+            self.value = "false"
+            if cores > 0:
+                self.value = "true"
+
+        return self
+
+    def to_cfn(self):
+        """
+        Return a tuple (cores_master,cores_compute) if disable_hyperthreading = true.
+
+        :return: string (cores_master,cores_compute)
+        """
+
+        cfn_params = {}
+
+        cluster_config = self.pcluster_config.get_section(self.section_key)
+        if cluster_config.get_param_value("disable_hyperthreading") == "true":
+            master_instance_type = cluster_config.get_param_value("master_instance_type")
+            compute_instance_type = cluster_config.get_param_value("compute_instance_type")
+            master_cores = get_instance_vcpus(self.pcluster_config.region, master_instance_type) // 2
+            compute_cores = get_instance_vcpus(self.pcluster_config.region, compute_instance_type) // 2
+            if master_cores < 0 or compute_cores < 0:
+                error(
+                    "Unable to get number of vcpus for {0} instance. ".format(
```

Comment:
I'd add an explanation about the configuration parameter causing the issue. 
In this way they could try to change the parameter and go ahead while waiting for an eventual fix.

Then what do you think about add the format at the end of the whole error message string and not in the middle?